### PR TITLE
Delete inactive Intercom users script

### PR DESCRIPTION
**Pull request by differentmatt**

### CODE REVIEW

File: scripts/deleteInactiveIntercomUsers.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        const deleteItems = [];
+        for (let i = 0; i < responses.length; i++) {
+          if (responses[i].body.conversations.length === 0) {
+            deleteItems.push({ delete: { user_id: oldUsers[i].user_id }});
+            // console.log(`DEBUG: deleting user ${oldUsers[i].id}`);
+          }
+        }
+        console.log(`DEBUG: deleting ${deleteItems.length} users`);
+        client.users.bulk(deleteItems)
+        .then((response) => {
```
Comment (basicer) [inline scripts/deleteInactiveIntercomUsers.js]: This seems weird.  If we move the setTimeout to another function, we can just return the promise here.

File: scripts/deleteInactiveIntercomUsers.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      const convoPromises = [];
+      for (const user of oldUsers) {
+        convoPromises.push(client.conversations.list({ type: 'user', intercom_user_id: user.id }));
+      }
+      Promise.all(convoPromises)
+      .then((responses) => {
+
+        // Bulk delete old users without conversations
+        const deleteItems = [];
+        for (let i = 0; i < responses.length; i++) {
```
Comment (basicer) [inline scripts/deleteInactiveIntercomUsers.js]: It seems like maybe Promise.map makes more sense here?

File: scripts/deleteInactiveIntercomUsers.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// NOTE: max 500 requests/minute, scroll closes after 1 minute and returns 100 user batches
+// So, scroll every 20s
+const scrollIntervalMilliseconds = 20 * 1000;
+
+const latestUpdate = new Date();
+latestUpdate.setUTCDate(latestUpdate.getUTCDate() - 30);
+
+client.users.scroll.each({}, function (response) {
+  console.log(`${new Date().toISOString()} DEBUG: scrolling ${scrollIntervalMilliseconds / 1000} seconds..`);
+  return new Promise((resolve) => {
```
Comment (basicer) [inline scripts/deleteInactiveIntercomUsers.js]: Maybe pull the timeout logic into a separate function.  Something like

```
function (timeout) { return (value) => new Promise((resove) => setTimeout(() => resolve(value), timeout)); }
```

And then you can do like `somePromise().then(wait(10)).then(...`

File: scripts/deleteInactiveIntercomUsers.js (hunk 1 above)
Comment (differentmatt) [inline scripts/deleteInactiveIntercomUsers.js]: It was less weird when I had debugging code before the resolve.  I'll move things around.

**Pull request closed without merging**